components/nimbus/src/tests/test_evaluator.rs

```diff
@@ -919,3 +926,231 @@ fn test_enrollment_bucketing() {
         }
     ));
 }
+
+#[test]
+fn test_events_sum_transform() {
+    let counter = MultiIntervalCounter::new(vec![SingleIntervalCounter::from(
+        IntervalData {
+            bucket_count: 3,
+            starting_instant: Utc::now(),
+            buckets: VecDeque::from(vec![1, 1, 0]),
+        },
+        IntervalConfig::new(3, Interval::Days),
+    )]);
+
+    let event_store = EventStore::from(vec![("app.foregrounded".to_string(), counter)]);
+    let mut targeting_attributes: TargetingAttributes = AppContext {
+        ..Default::default()
+    }
+    .into();
+    targeting_attributes.event_store = Some(event_store);
+    assert_eq!(
+        targeting(
+            "'app.foregrounded'|eventsSum('Days', 3, 0) > 2",
+            &targeting_attributes
+        ),
+        Some(EnrollmentStatus::NotEnrolled {
+            reason: NotEnrolledReason::NotTargeted
+        })
+    );
+    assert_eq!(
+        targeting(
+            "'app.foregrounded'|eventsSum('Days', 3, 0) > 1",
+            &targeting_attributes
+        ),
+        None
+    );
+}
+
+#[test]
+fn test_events_count_non_zero_transform() {
+    let counter = MultiIntervalCounter::new(vec![SingleIntervalCounter::from(
+        IntervalData {
+            bucket_count: 3,
+            starting_instant: Utc::now(),
+            buckets: VecDeque::from(vec![1, 2, 0]),
+        },
+        IntervalConfig::new(3, Interval::Days),
+    )]);
+
+    let event_store = EventStore::from(vec![("app.foregrounded".to_string(), counter)]);
+    let mut targeting_attributes: TargetingAttributes = AppContext {
+        ..Default::default()
+    }
+    .into();
+    targeting_attributes.event_store = Some(event_store);
+    assert_eq!(
+        targeting(
+            "'app.foregrounded'|eventsCountNonZero('Days', 3, 0) > 2",
+            &targeting_attributes
+        ),
+        Some(EnrollmentStatus::NotEnrolled {
+            reason: NotEnrolledReason::NotTargeted
+        })
+    );
+    assert_eq!(
+        targeting(
+            "'app.foregrounded'|eventsCountNonZero('Days', 3, 0) > 1",
+            &targeting_attributes
+        ),
+        None
+    );
+}
+
+#[test]
+fn test_events_average_per_interval_transform() {
+    let counter = MultiIntervalCounter::new(vec![SingleIntervalCounter::from(
+        IntervalData {
+            bucket_count: 7,
+            starting_instant: Utc::now(),
+            buckets: VecDeque::from(vec![1, 2, 0, 0, 0, 2, 3]),
+        },
+        IntervalConfig::new(7, Interval::Days),
+    )]);
+
+    let event_store = EventStore::from(vec![("app.foregrounded".to_string(), counter)]);
+    let mut targeting_attributes: TargetingAttributes = AppContext {
+        ..Default::default()
+    }
+    .into();
+    targeting_attributes.event_store = Some(event_store);
+    assert_eq!(
+        targeting(
+            "'app.foregrounded'|eventsAveragePerInterval('Days', 7, 0) > 2",
```

Comment:
Do we need a floating point test of this too?

Reply:
Yep!

Reply:
Just double checking -- are you suggesting a test where the right side of the operation is a floating point value?

Reply:
Yes, since we have changed this to a float, we might as well expose it here, right?

Reply:
Yep! Just wasn't sure if you were suggesting I test it in the transform (it'll just convert it to an int)